baybe/recommenders/pure/nonpredictive/clustering.py

```diff
@@ -54,6 +43,11 @@ class SKLearnClusteringRecommender(NonPredictiveRecommender, ABC):
     """Optional model parameter that will be passed to the surrogate constructor.
     This is initialized with reasonable default values for the derived child classes."""
 
+    @staticmethod
+    @abstractmethod
+    def _get_model_cls() -> type[ClusterMixin]:
```

Comment:
Why is this private? The previous class variable was public, so why the change?

Reply:
Yes, in general I'd like to be a bit more cautious what we expose as public and what not. In the past, pretty much everything was public (like stuff in this module, which was written long time ago), even though not really needed. That only causes pain with backward compatibility when you want to refactor stuff. Since the method is really not expected to be used outside, it think its better to keep it private for now. We can always change when needed: making private stuff public is easy, the other way round not. So I took this opportunity....